modules/consensus/synchronize.go

```diff
@@ -355,3 +400,73 @@ func (cs *ConsensusSet) threadedReceiveBlock(id types.BlockID) modules.RPCFunc {
 	}
 	return managedFN
 }
+
+// threadedInitialBlockchainDownload performs the IBD on outbound peers. Blocks
+// are downloaded from one peer at a time in 5 minute intervals, so as to
+// prevent any one peer from significantly slowing down IBD.
+//
+// NOTE: IBD will succeed right now when each peer has a different blockchain.
+// The height and the block id of the remote peers' current blocks are not
+// checked to be the same. This can cause issues if you are connected to
+// outbound peers <= v0.5.1 that are stalled in IBD.
```

Comment:
if you have somehow gotten 5 outbound peers which are all stalled...

hmm. Maybe peers should always return `more available` until they've managed to not be stalled themselves. Of course, that opens them up to a deadlock situation where all the peers connected to eachother are stalled, but then the peers will fix this by booting eachother.

So, I'm in favor of always returning `more available` until you are caught up. This could cause weird edge case behavior though, which would mean more testing...

heh